planner/core/find_best_task.go

```diff
@@ -1635,6 +1635,12 @@ func (ds *DataSource) convertToIndexScan(prop *property.PhysicalProperty,
 	}
 	if candidate.isMatchProp {
 		cop.keepOrder = true
+		if cop.tablePlan != nil && !ds.tableInfo.IsCommonHandle {
+			col, isNew := cop.tablePlan.(*PhysicalTableScan).appendExtraHandleCol(ds)
+			cop.extraHandleCol = col
+			cop.needExtraProj = cop.needExtraProj || isNew
```

Comment:
Does this code related to this issue?

Reply:
Yes, we should make sure `model.ExtraPhysTblID` in the last column. So move the code to the front.